src/translator/batching_pool.cpp

```diff
@@ -24,11 +34,11 @@ size_t BatchingPool::generateBatch(Batch &batch) {
   batch.clear();
   size_t paddedBatchSize = 0;
 
-  for (size_t length = 0; length < bucket_.size(); length++) {
+  for (size_t length = 0; length <= maxActiveBucketLength_; length++) {
```

Comment:
Can this be reverted back to `bucket_.size()`?

Reply:
There's `maxLengthBreak_ + pivotSlack` sizing initially. `bucket_.size()` could be `1.5*maxLengthBreak_` looping a bit more although empty. It's not structured and one more point of error, I don't like it much either, but....